drv/Map.drv

```diff
@@ -823,6 +823,41 @@ public interface MAP KEY_VALUE_GENERIC extends FUNCTION KEY_VALUE_GENERIC, Map<K
 
 #endif
 
+	/** Returns an unmodifiable, type-specific map with the specified entries.
+	 * See {@link java.util.Map#ofEntries(Map.Entry[])} */
+#if !KEYS_PRIMITIVE || !VALUES_PRIMITIVE
+	@SafeVarargs
+	@SuppressWarnings({"varargs", "unchecked"})
+#else
+	@SuppressWarnings("varargs")
+#endif
+	static KEY_VALUE_GENERIC MAP KEY_VALUE_GENERIC ofEntries(MAP.Entry KEY_VALUE_GENERIC ... entries) {
+		if (entries.length == 0) return MAPS.EMPTY_MAP;
+		if (entries.length == 1) return MAPS.singleton(entries[0].ENTRY_GET_KEY(), entries[0].ENTRY_GET_VALUE());
+
+		KEY_GENERIC_TYPE[] keys = KEY_GENERIC_ARRAY_CAST new KEY_TYPE[entries.length];
+		VALUE_GENERIC_TYPE[] vals = VALUE_GENERIC_ARRAY_CAST new VALUE_TYPE[entries.length];
+
+		for (int i = 0; i < entries.length; ++i) {
+			MAP.Entry KEY_VALUE_GENERIC entry = entries[i];
+			keys[i] = entry.ENTRY_GET_KEY();
+			// Prevent duplicate keys
+			for (int j = 0; j < i; ++j) if (KEY_EQUALS(keys[j], keys[i])) throw new IllegalArgumentException("duplicate key: " + keys[i]);
+			vals[i] = entry.ENTRY_GET_VALUE();
+		}
+		if (entries.length <= 8) {
+			return MAPS.unmodifiable(new ARRAY_MAP KEY_VALUE_GENERIC_DIAMOND(keys, vals, entries.length));
+		} else {
+			return MAPS.unmodifiable(new OPEN_HASH_MAP KEY_VALUE_GENERIC_DIAMOND(keys, vals, 0.75f));
```

Comment:
this creates the keys / values array unnecessarily. I think the code should be like
```
if (number of entries is small)
  create keys and vals as above (with duplicate checking)
  return array map with keys and vales
else
  create appropriately sized hash map
  for each entry:
    if map.put(entry.key, entry.value) != null then throw exception
  return hash map
```

Reply:
you are quite right, let me do that

Reply:
would using `contains` instead of the return value of `#put` be better? For the edge case where the previous value is `null`